cmd/restic/integration_test.go

```diff
@@ -1,3 +1,72 @@
-// build +integration
+// +build integration
 
-package main_test
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	. "github.com/restic/restic/test"
+)
+
+var TestDataFile = flag.String("test.datafile", "", `specify tar.gz file with test data to backup and restore (required for integration test)`)
+
+func cmd_init(t testing.TB) {
+	cmd := &CmdInit{}
+	OK(t, cmd.Execute(nil))
+
+	t.Logf("repository initialized at %v", opts.Repo)
+}
+
+func cmd_backup(t testing.TB, target []string) {
+	cmd := &CmdBackup{}
+	t.Logf("backing up %v", target)
+
+	OK(t, cmd.Execute(target))
+}
+
+func system(command string, args ...string) error {
+	cmd := exec.Command(command, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	return cmd.Run()
+}
+
+func TestBackup(t *testing.T) {
+	if *TestDataFile == "" {
+		t.Fatal("no data tar file specified, use flag '-test.datafile'")
+	}
+
+	tempdir, err := ioutil.TempDir(*TestTempDir, "restic-test-")
+	OK(t, err)
+
+	defer func() {
+		if !*TestCleanup {
+			t.Logf("leaving temporary directory %v used for test", tempdir)
+			return
+		}
+
+		OK(t, os.RemoveAll(tempdir))
+	}()
+
+	// use cache dir within tempdir
+	OK(t, os.Setenv("RESTIC_CACHE", filepath.Join(tempdir, "cache")))
+
+	// configure environment
+	opts.Repo = filepath.Join(tempdir, "repo")
+	OK(t, os.Setenv("RESTIC_PASSWORD", *TestPassword))
+	cmd_init(t)
```

Comment:
seems like everything in this function above this line doesn't really belong into this test since it's very generic, not specific to what is being tested here, and we will do the same thing for lots of other tests... maybe we can hide this behind some kind of setup function?